echopop/inversion/inversion_matrix.py

```diff
@@ -0,0 +1,1230 @@
+    reference_nasc = reference_nasc.reset_index().set_index(idx)
+
+    # Concatenate the parameters
+    return pd.concat(
+        [
+            reference_nasc,
+            parameters.reset_index()
+            .drop("number_density", axis=1)
+            .set_index(idx)
+            .reindex(reference_nasc.index),
+        ],
+        axis=1,
+    )
+
+
+def invert_transect_number_density(
+    acoustic_data: pd.DataFrame,
+    reference_nasc: pd.DataFrame,
+    parameters: pd.DataFrame,
+):
+    """
+    Compute areal number density aggregated by survey transect.
+
+    This function estimates the areal number density of organisms by
+    weighting volumetric densities by NASC contributions and aggregating
+    across survey transects. The method accounts for varying layer
+    thickness and spatial sampling intensity.
+
+    Parameters
+    ----------
+    acoustic_data : pd.DataFrame
+        DataFrame containing acoustic measurements including:
+        - nasc: Nautical Area Scattering Coefficient values
+        - thickness_mean: Mean layer thickness in meters
+    reference_nasc : pd.DataFrame
+        DataFrame with NASC values at reference frequency for weighting
+    parameters : pd.DataFrame
+        DataFrame containing biological parameters including:
+        - number_density: Volumetric number density (individuals/m³)
+
+    Returns
+    -------
+    pd.Series
+        Areal number density in individuals/m² aggregated by transect,
+        converted from nautical miles to meters (factor of 1852²)
+
+    Notes
+    -----
+    The areal number density calculation follows:
+
+    .. math::
+        N_A = N_V \\cdot w \\cdot h \\cdot n \\cdot 1852^2
+
+    where N_A is areal density, N_V is volumetric density, w is NASC weight,
+    h is layer thickness, n is interval count, and 1852² converts from
+    nautical miles² to meters².
+
+    NASC weighting ensures proper spatial representation:
+
+    .. math::
+        w = \\frac{\\text{NASC}_{ref}}{\\text{NASC}_{acoustic}}
+    """
+
+    # Create copies
+    acoustic_df = acoustic_data.copy()
+    reference_df = reference_nasc.copy()
+
+    # Find shared indices
+    idx = list(set(reference_nasc.index.names).intersection(acoustic_data.index.names))
+
+    # Reset the indices for each
+    acoustic_df = acoustic_df.reset_index().set_index(idx)
+    reference_df = reference_df.reset_index().set_index(idx)
+
+    # Compute the NASC weights
+    reference_df["nasc_weight"] = reference_df["nasc"] / acoustic_df["nasc"]
+
+    # Get the counts of each transect
+    interval_counts = reference_df.groupby(level="transect_num")["nasc"].count()
+
+    # Compute the areal number density
+    areal_number_density = (
+        parameters["number_density"]
+        * reference_df["nasc_weight"]
+        * acoustic_df["thickness_mean"]
+        * interval_counts
+    ).fillna(0.0) * 1852**2
+
+    # Return
+    return areal_number_density
+
+
+def invert_intervals_number_density(
```

Comment:
```suggestion
def calculate_intervals_number_density(
```
since it is not an inversion.